src/libostree/ostree-sysroot-upgrader.c

```diff
@@ -616,8 +617,13 @@ ostree_sysroot_upgrader_deploy (OstreeSysrootUpgrader *self, GCancellable *cance
         return FALSE;
 
       if (!ostree_sysroot_simple_write_deployment (self->sysroot, self->osname, new_deployment,
-                                                   self->merge_deployment, 0, cancellable, error))
+                                                   self->merge_deployment, write_deployment_flags, cancellable, error))
         return FALSE;
+
+
+      if ((self->flags & OSTREE_SYSROOT_UPGRADER_FLAGS_KEXEC) > 0)
```

Comment:
Very minor style thing, having a `{` is preferred on at least the outer conditional in these nested if setups.